NuRadioReco/modules/channelGalacticNoiseAdder.py

```diff
@@ -0,0 +1,209 @@
+            d_f = freqs[2] - freqs[1]
+            sampling_rate = channel.get_sampling_rate()
+            channel_spectrum = channel.get_frequency_spectrum()
+            passband_filter = (freqs > passband[0]) & (freqs < passband[1])
+            noise_spec_sum = np.zeros_like(channel.get_frequency_spectrum())
+            flux_sum = np.zeros(freqs[passband_filter].shape)
+            efield_sum = np.zeros((3, freqs.shape[0]), dtype=np.complex)
+            if self.__debug:
+                plt.close('all')
+                fig = plt.figure(figsize=(12, 8))
+                ax_1 = fig.add_subplot(221)
+                ax_2 = fig.add_subplot(222)
+                ax_3 = fig.add_subplot(223)
+                ax_4 = fig.add_subplot(224)
+                ax_1.grid()
+                ax_1.set_yscale('log')
+                ax_2.grid()
+                ax_2.set_yscale('log')
+                ax_3.grid()
+                ax_3.set_yscale('log')
+                ax_4.grid()
+                ax_4.set_yscale('log')
+                ax_1.set_xlabel('f [MHz]')
+                ax_2.set_xlabel('f [MHz]')
+                ax_3.set_xlabel('f [MHz]')
+                ax_4.set_xlabel('f [MHz]')
+                ax_1.set_ylabel('T [K]')
+                ax_2.set_ylabel('S [W/m²/MHz]')
+                ax_3.set_ylabel('E [V/m]')
+                ax_4.set_ylabel('U [V]')
+                ax_4.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), c='C0')
+                ax_4.set_ylim([1.e-8, None])
+                fig1 = plt.figure()
+                ax1 = fig1.add_subplot(211)
+                ax2 = fig1.add_subplot(212)
+                ax1.plot(channel.get_times(), channel.get_trace(), label='original trace')
+                ax2.plot(channel.get_frequencies() / units.MHz, np.abs(channel.get_frequency_spectrum()), label='original spectrum')
+                ax1.grid()
+                ax2.grid()
+            for i_pixel in range(healpy.pixelfunc.nside2npix(self.__n_side)):
+                azimuth = local_coordinates[i_pixel].az.rad
+                zenith = np.pi / 2. - local_coordinates[i_pixel].alt.rad
+                if zenith > 90. * units.deg:
+                    continue
+                temperature_interpolator = scipy.interpolate.interp1d(self.__interpolaiton_frequencies, np.log10(noise_temperatures[:, i_pixel]), kind='quadratic')
+                noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
+                S = (2. * scipy.constants.Boltzmann * (freqs[passband_filter] / units.Hz)**2 / scipy.constants.c**2 * (noise_temperature / units.kelvin) * solid_angle) * (units.watt / units.m**2 / units.Hz)
+                S[np.isnan(S)] = 0
+                S_per_bin = S * d_f
+                flux_sum += S_per_bin
+                E = np.sqrt(S_per_bin / (units.watt / units.m**2) / (scipy.constants.c * scipy.constants.epsilon_0)) / (d_f) * units.V / units.m
```

Comment:
I think a bit more in-code documentation would be good. For example what physical quantity is S, and what do you do to convert it to an electric field. 
Also the application of the units seems wrong, or at leas not the way it should go. Currently, the units are explicitely introduced. Instead, the scipy constants should be multiplied by its units. Then the rest of the calculation can be done without explicit specification of units. 
so e.g. `speed_of_light = scipy.constants.c * units.m/ units.s`. Same for the other constants. 
Then l166 would just be
`E = np.sqrt(S_per_bin / (units.watt ) / (speed_of_light * epsilon_0)) / (d_f)`
Same for the line where you calculate `S`.

Reply:
Done